akka-actor/src/main/scala/akka/actor/dungeon/Dispatch.scala

```diff
@@ -147,10 +149,15 @@ private[akka] trait Dispatch { this: ActorCell ⇒
     unwrappedMessage match {
       case _: NoSerializationVerificationNeeded ⇒ envelope
       case msg ⇒
-        val deserializedMsg = serializeAndDeserializePayload(msg)
-        envelope.message match {
-          case dl: DeadLetter ⇒ envelope.copy(message = dl.copy(message = deserializedMsg))
-          case _              ⇒ envelope.copy(message = deserializedMsg)
+        Try(serializeAndDeserializePayload(msg)) match {
+          case Success(deserializedMsg) ⇒
+            envelope.message match {
+              case dl: DeadLetter ⇒ envelope.copy(message = dl.copy(message = deserializedMsg))
+              case _              ⇒ envelope.copy(message = deserializedMsg)
+            }
+          case Failure(e) ⇒
+            throw new Exception(s"Failed to serialize and deserialize message of type ${msg.getClass.getName} for testing. " +
```

Comment:
rather RuntimeException, for java

I agree that it might be worth adding another exception type for this